### ISSUE TITLE
error  Parsing error: 'import' and 'export' may appear only with 'sourceType: module'

### ISSUE THREAD

**Issue by gforceg** [eslint/eslint#10397]
**Tell us about your environment**
Fedora release 28 (Twenty Eight)
npm 6.0.0
node v8.4.0
eslint v4.19.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default
**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    "extends": "google",
    "env": {
        "es6": true
    }
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

module-a.js
```js

export function doA() { console.log('a!') }

```
module-b.js
```js

export function doB() { console.log('b!') }

```
index.js
```js
import { doA } from './module-a.js'
import { doB } from './module-b.js'

doA();
doB()
```

```bash
npx eslint ./*.js
```

**What did you expect to happen?**
I thought the linter would report problems w/ my use of semicolons.

**What actually happened? Please include the actual, raw output from ESLint.**
It reported that it could not parse 'import' and 'export' due to my configuration.
action@localhost learning-eslint] $ npx eslint ./*.js

/home/action/src/my-stuff/learning/learning-eslint/index.js
  1:1  error  Parsing error: 'import' and 'export' may appear only with 'sourceType: module'

/home/action/src/my-stuff/learning/learning-eslint/module-a.js
  1:1  error  Parsing error: 'import' and 'export' may appear only with 'sourceType: module'

/home/action/src/my-stuff/learning/learning-eslint/module-b.js
  1:1  error  Parsing error: 'import' and 'export' may appear only with 'sourceType: module'

✖ 3 problems (3 errors, 0 warnings)

I changed my configuration to the following:
<details>
<summary>Configuration 2</summary>
```js
module.exports = {
    "extends": "google",
    "env": {
        "es6": true
    },
    "sourceType": "module"
}
```
</details>

Now eslint explains that my config is invalid.:

```bash
[action@localhost learning-eslint] $ npx eslint ./*.js
ESLint configuration in /home/action/src/my-stuff/learning/learning-eslint/.eslintrc.js is invalid:
        - Unexpected top-level property "sourceType".
```
From what I've read "sourceType" is an older configuration key that is no longer used in .eslintrc.

**Comment by gforceg** [comment]
I switched from `eslint-config-google` to `eslint-plugin-import` and that seems to have solved the problem.

**Comment by not-an-aardvark** [comment]
Hi, I'm glad you ended up getting something working. In case you want to switch back to `eslint-config-google`, the correct way to use `sourceType` is to include it in a `parserOptions` object, rather than putting it at the top level of your config:

```
{
    "parserOptions": {
        "sourceType": "module"
    }
}
```

**Comment by gforceg** [comment]
Thank you @not-an-aardvark 
I was about to reopen because I switched to "eslint:recommended" and the issue persisted.
